jahorwitz/slopopedia
# Feature/slop 5/search filters

## Describe your changes

Added search page with full functionality for all filters described in the Figma design.

**NOTE:** There are issues with the `<MovieCard/>` and `<MovieCardList/>` components that are outside of the scope of this ticket. These include issues with the styling of the movie cards and an issue with the keys on movie cards potentially being duplicates. The issue with the movie cards could also have something to do with the create-movies script.

### Testing Instructions
1. From the frontend directory, switch to the search filters branch (`git switch feature/slop-5/search-filters`)
2. `git pull`
3.  Seed local DB with movies and keywords using `BACKEND/scripts/create-movies.js` (see below section if you need help with that)
4. Navigate to search page and see if stuff breaks.


#### How to seed the DB with keywords and movies
1. Open `BACKEND/Scripts/create-movies.js` and replace `https://slopopedia-api-a5fe9aef64e8.herokuapp.com/api/graphql` with `http://localhost:8080/api/graphql` (`Ctrl + h` in vscode)
2. Ensure that your `.env` file contains your username (`ADMIN_PANEL_USERNAME`) and password (`ADMIN_PANEL_PASSWORD`) for the keystone admin panel. 
3. Add `status: "published"` around line 100 so the movies will show up.
4. Ensure that the backend is running.
5. From the backend directory run the create movies script using `node ./scripts/create-movies/create-movies.js`
6. The terminal should output that it's successfully populated the db with keywords and movies.

### Library changes
- Updated Headless UI
- Added Fuse.js for handling filtering movies by string.

## Issue ticket number and link

https://tripleten-apiary.atlassian.net/browse/SLOP-5?atlOrigin=eyJpIjoiMzcwZDU5Y2FjOGUzNGVjM2JkOGJhNWQwNDNiNGQ1MjQiLCJwIjoiaiJ9

## Checklist before requesting a review

- [x] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
